lib/swoosh/mailer.ex

```diff
@@ -82,14 +87,21 @@ defmodule Swoosh.Mailer do
     end
   end
 
-  def deliver(_adapter, %Swoosh.Email{from: nil}, _config) do
+  def deliver(%Swoosh.Email{from: nil}, _config) do
     {:error, :from_not_set}
   end
-  def deliver(_adapter, %Swoosh.Email{from: {_name, address}}, _config) when address in ["", nil] do
+  def deliver(%Swoosh.Email{from: {_name, address}}, _config)
+      when address in ["", nil] do
     {:error, :from_not_set}
   end
-  def deliver(adapter, %Swoosh.Email{} = email, config) do
-    config = Swoosh.Mailer.parse_runtime_config(config)
+  def deliver(%Swoosh.Email{} = email, {otp_app, mailer, module_config, config}) do
+    config =
```

Comment:
Probably worth extracting this into its own function, `parse_runtime_config` maybe?